library/std/src/sys/unix/fs.rs

```diff
@@ -95,6 +95,60 @@ use libc::{dirent64, fstat64, ftruncate64, lseek64, lstat64, off64_t, open64, st
 
 pub use crate::sys_common::fs::try_exists;
 
+pub(crate) mod apis {
+    use crate::io;
+    use crate::path::PathBuf;
+    use crate::sys::fs;
+    pub(crate) use crate::sys::fs::{
+        DirBuilder, DirEntry, File, FileAttr, FilePermissions, FileTimes, FileType, OpenOptions,
+        ReadDir,
+    };
+    use crate::sys::path::PathLike;
+
+    pub(crate) fn remove_file<P: PathLike>(path: P) -> io::Result<()> {
+        path.with_path(fs::unlink)
```

Comment:
So is the only remaining work after this PR to swap these to `with_native_path`? That's exciting!

Reply:
Yeah, that's the idea! It may not be trivial in all cases though so I've left that for follow up PRs.